app/main.py

```diff
@@ -1,2 +1,130 @@
+import copy
+
+
+class DictionaryMember:
+    def __init__(
+            self,
+            key: int | str | tuple | float,
+            value: any,
+            index: int = None
+    ) -> None:
+        self.key = key
+        self.value = value
+        self.index = index
+
+
 class Dictionary:
-    pass
+
+    def __init__(
+            self,
+            capacity: int = 10,
+            load_factor: float = 0.7
+    ) -> None:
+
+        self.size = 0
+        self.capacity = capacity
+        self.load_factor = load_factor
+        self.table = [None] * self.capacity
+
+    def _index(
+            self,
+            key: int | str | tuple | float
+    ) -> int:
+
+        index = hash(key) % self.capacity
+
+        if (
+                self.table[index] is None
+                or self.table[index].key == key
+        ):
+            return index
+
+        else:
+            try:
+                while (
+                        self.table[index] is not None
+                        and self.table[index].key != key
+                ):
+                    index += 1
+                    if index > len(self.table):
+                        index = 0
+                return index
+            except IndexError:
+                index = 0
+                while (
+                        self.table[index] is not None
+                        and self.table[index].key != key
+                ):
+                    index += 1
+                    if index > len(self.table):
+                        index = 0
```

Comment:
Why?

Reply:
Because I'm highly educated person and can't count to 9. =)))